src/main/java/badgpt/util/TaskList.java

```diff
@@ -0,0 +1,184 @@
+package badgpt.util;
+
+import badgpt.exceptions.ListEmptyException;
+import badgpt.exceptions.ListFullException;
+import badgpt.exceptions.SameStatusException;
+import badgpt.exceptions.TaskNotFoundException;
+
+import badgpt.tasks.Deadline;
+import badgpt.tasks.Event;
+import badgpt.tasks.Task;
+import badgpt.tasks.ToDo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a todo list.
+ */
+public class TaskList {
+    private List<Task> tasks = new ArrayList<>(100);
+    private TasksUi tasksUi = new TasksUi();
+    private boolean hasChanges = false;
+
+    /**
+     * Stores a Task object in the list. The list has a max size of 100.
+     *
+     * @param task The Task object to be stored in the list.
+     * @throws ListFullException If the list already has 100 tasks.
+     */
+    public void store(Task task) throws ListFullException {
```

Comment:
would it be better to give a clearer, verb form of the method name, such as storeTask()?
```suggestion
    public void storeTask(Task task) throws ListFullException {
```